modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptPropertyDefValGenerator.cs

```diff
@@ -19,49 +19,54 @@ public void Execute(GeneratorExecutionContext context)
             if (context.AreGodotSourceGeneratorsDisabled())
                 return;
 
-            INamedTypeSymbol[] godotClasses = context
-                .Compilation.SyntaxTrees
-                .SelectMany(tree =>
-                    tree.GetRoot().DescendantNodes()
-                        .OfType<ClassDeclarationSyntax>()
-                        .SelectGodotScriptClasses(context.Compilation)
-                        // Report and skip non-partial classes
-                        .Where(x =>
+            MarshalUtils.TypeCache? typeCache = null;
+            // Not sure if this is needed to replicate the previous Distinct behavior.
+            // Is it ever allowed to see the same class in multiple syntax trees?
+            var processedClasses = new HashSet<INamedTypeSymbol>();
+            foreach (var tree in context.Compilation.SyntaxTrees)
```

Comment:
I did that, but I had to raise the target framework from netstandard2.0 to net6.0 to get access to DistinctBy. I believe this is the default for C# projects so I don't think this should be an issue, but if need be we can always implement DistinctBy ourselves and go back to netstandard2.0.

Reply:
Oh, I didn't realize `DistinctBy` is .NET 6.0+ only. We can't raise the target to `net6.0` though because source generators must target `netstandard2.0`.

Reply:
Ah, I'll put that back then. I think going back to the loop also lets me avoid re-generating the SemanticModel anyways.

Reply:
I can't think properly right now, but this part doesn't seem right. You're iterating over all syntax trees looking for godot classes, but you only process the properties for the first syntax tree that finds a godot class. This syntax tree may not be the one containing a property/field initializer we need. In fact, a class could be divided into multiple syntax trees (with `partial`), each one declaring properties with initializers.

If all of this is to get the semantic model, you could get it inline from the property initializer. See my other review comment.